testdata/go/v2/memcached-operator/Dockerfile

```diff
@@ -15,7 +17,7 @@ COPY api/ api/
 COPY controllers/ controllers/
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=$GOARCH GO111MODULE=on go build -a -o manager main.go
 
```

Comment:
/hold

See that you are changing the testdata samples. 
It is not the code used to generate the projects. Its implementation lives on Kubebuilder's repositiory.
Also, why you are updating the go/v2 ?

Reply:
What is the goal of this PR?
What would you like to achieve?

Reply:
We are working on the support of operator-sdk on ppc64le/s390x. So we need to fix issues preventing to build the commands and images, and run the tests managed by the operator-sdk Makefile on these arches. We also need to check that the generated Ansible/Helm/Go operator examples can be used on ppc64/s390x.
Currently the GO operator examples generate Go images that includes an amd64 manager command.
This makes the e2e operator-sdk tests failing on ppc64le/s390x arches and the Dockerfile of the Go projects generated by `operator-sdk init` has to be fixed to generate a valid ppc64le or s390x image.
The goal of this PR was to fix the Go Makefile/Dockerfile to generate operator sample images with the correct architecture.

Reply:
@gquillar 

> We are working on the support of operator-sdk on ppc64le/s390x. So we need to fix issues preventing building the commands and images and run the tests managed by the operator-sdk Makefile on these arches.

Could you please raise an issue and add the errors faced when they are faced?

> We also need to check that the generated Ansible/Helm/Go operator examples can be used on ppc64/s390x.

If the goal is to make the tool to support these architectures. 
That does not seem the right direction.

The Dockerfile only will build the Operator/Manager image and the projects use other images as well by default. 

Therefore, note that you are changing examples which are generated with the tool and not any code that is used to generate it. Yes, if we want to change the Dockerfile of the Golang scaffold that must be done in Kubebuilder. Also, I do not think that we should change the go/v2 since it is a legacy plugin and is not used by default.
 
So, WDYT? Can we begin with an issue in the SDK repo adding all scenarios, motivations, issues and steps for we to reproduce then? So that we can discuss it?

Reply:
HI @gquillar, 

if you want to change the Dockerfile for further releases, then the change need to be done in the templates,

For Golang, it must be done in: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/golang/v3/scaffolds/internal/templates/dockerfile.go

For Ansible/Helm and any other language plugin, we need to update the Dockerfile of the specific plugin.
In this way, I'd recommend you push a PR against Kubebuilder, and then after that gets merged, we need to update all language plugins inside of SDK and those used by it and have the code on other repos.